cardano-db-sync/src/Cardano/DbSync/Api.hs

```diff
@@ -199,52 +200,61 @@ getPrunes :: SyncEnv -> Bool
 getPrunes = do
   DB.pcmPruneTxOut . getPruneConsume
 
-fullInsertOptions :: Bool -> InsertOptions
-fullInsertOptions useLedger =
-  InsertOptions
-    { ioInOut = True
-    , ioUseLedger = useLedger
-    , ioShelley = True
-    , ioRewards = True
-    , ioMultiAssets = True
-    , ioMetadata = True
-    , ioKeepMetadataNames = Strict.Nothing
-    , ioPlutusExtra = True
-    , ioOffChainPoolData = True
-    , ioGov = True
+extractInsertOptions :: SyncPreConfig -> SyncInsertOptions
+extractInsertOptions cfg =
+  case pcInsertConfig cfg of
+    FullInsertOptions -> fullInsertOptions
+    OnlyUTxOInsertOptions -> onlyUTxOInsertOptions
+    OnlyGovInsertOptions -> onlyGovInsertOptions
+    DisableAllInsertOptions -> disableAllInsertOptions
+    SyncInsertConfig opts -> opts
+
+fullInsertOptions :: SyncInsertOptions
+fullInsertOptions =
+  SyncInsertOptions
+    { sioTxOut = TxOutEnable
+    , sioLedger = LedgerEnable
+    , sioShelley = ShelleyEnable
+    , sioMultiAsset = MultiAssetEnable
+    , sioMetadata = MetadataEnable
```

Comment:
is the consensus that in the presets a user won't be able to add whitelists?

Reply:
As we discussed earlier, a preset will overwrite _all_ other settings, for the time being